bson/src/main/org/bson/types/ObjectId.java

```diff
@@ -224,9 +200,12 @@ private ObjectId(final int timestamp, final int machineIdentifier, final short p
             throw new IllegalArgumentException("The counter must be between 0 and 16777215 (it must fit in three bytes).");
         }
         this.timestamp = timestamp;
-        this.machineIdentifier = machineIdentifier;
-        this.processIdentifier = processIdentifier;
         this.counter = counter & LOW_ORDER_THREE_BYTES;
+        this.randomValue[0] = int2(machineIdentifier);
```

Comment:
Make sure this is adequately covered by regression tests to ensure that it round trips exactly the same is it does now.

Reply:
The 5-byte random value is now represented as two integers, so the round trips should be the same as it was when the machine and process IDs were used.